lua/core/info.lua

```diff
@@ -136,67 +57,89 @@ function M.toggle_popup(ft)
   local client_name = ""
   local client_id = 0
   local document_formatting = false
-  local num_caps = 0
   if client ~= nil then
     is_client_active = not client.is_stopped()
     client_enabled_caps = require("lsp").get_ls_capabilities(client.id)
-    num_caps = vim.tbl_count(client_enabled_caps)
     client_name = client.name
     client_id = client.id
     document_formatting = client.resolved_capabilities.document_formatting
   end
 
-  local buf_lines = {}
-  vim.list_extend(buf_lines, M.banner)
-
   local header = {
-    indent .. "Detected filetype:     " .. tostring(ft),
-    indent .. "Treesitter active:     " .. tostring(next(vim.treesitter.highlighter.active) ~= nil),
-    "",
+    fmt("Detected filetype:      %s", ft),
+    fmt("Treesitter active:      %s", tostring(next(vim.treesitter.highlighter.active) ~= nil)),
   }
-  vim.list_extend(buf_lines, header)
 
+  local text = require "interface.text"
   local lsp_info = {
-    indent .. "Language Server Protocol (LSP) info",
-    indent .. "* Associated server:   " .. client_name,
-    indent .. "* Active:              " .. tostring(is_client_active) .. " (id: " .. tostring(client_id) .. ")",
-    indent .. "* Supports formatting: " .. tostring(document_formatting),
-    indent .. "* Capabilities list:   " .. table.concat(vim.list_slice(client_enabled_caps, 1, num_caps / 2), ", "),
-    indent .. indent .. indent .. table.concat(vim.list_slice(client_enabled_caps, ((num_caps / 2) + 1)), ", "),
-    "",
+    "Language Server Protocol (LSP) info",
+    fmt("* Associated server:    %s", client_name),
+    fmt("* Active:               %s (id: %d)", tostring(is_client_active), client_id),
+    fmt("* Supports formatting:  %s", tostring(document_formatting)),
   }
-  vim.list_extend(buf_lines, lsp_info)
+  local caps_text = "* Capabilities list:    "
+  local caps_text_len = caps_text:len()
+  local enabled_caps = text.format_table(client_enabled_caps, 3, " | ")
+  enabled_caps = text.shift_left(enabled_caps, caps_text_len)
+  enabled_caps[1] = fmt("%s%s", caps_text, enabled_caps[1]:sub(caps_text_len + 1))
```

Comment:
if you mistakenly open it in terminal 😅 
<img width="945" alt="Screen Shot 2021-08-16 at 3 39 35 PM" src="https://user-images.githubusercontent.com/10992695/129554742-6793d7ff-6c72-490c-8104-768d7d49fd22.png">

Reply:
also if i open a `test.fish` file and run the same command 😢